File: apps/cli/internal/client/jobs.go
package client

type Job struct {
	ID                string  `json:"id"`
	Name              string  `json:"name"`
	IsCron            bool    `json:"is_cron"`
	Image             string  `json:"image"`
	Command           string  `json:"command"`
	Schedule          string  `json:"schedule"`
	ConcurrencyPolicy string  `json:"concurrency_policy"`
	HistoryLimit      int     `json:"history_limit"`
	Status            string  `json:"status"`
	LastRunAt         *string `json:"last_run_at"`
}

type JobRun struct {
	ID         string  `json:"id"`
	Status     string  `json:"status"`
	Log        string  `json:"log"`
	StartedAt  *string `json:"started_at"`
	FinishedAt *string `json:"finished_at"`
}

type CreateJobBody struct {
	Name              string  `json:"name"`
	IsCron            bool    `json:"is_cron"`
	Image             string  `json:"image"`
	Command           string  `json:"command,omitempty"`
	Schedule          string  `json:"schedule,omitempty"`
	ConcurrencyPolicy string  `json:"concurrency_policy,omitempty"`
	HistoryLimit      int     `json:"history_limit,omitempty"`
	CPURequest        string  `json:"cpu_request,omitempty"`
	CPULimit          string  `json:"cpu_limit,omitempty"`
	MemoryRequest     string  `json:"memory_request,omitempty"`
	MemoryLimit       string  `json:"memory_limit,omitempty"`
	EnvVars           string  `json:"env_vars,omitempty"`
	NodeID            *string `json:"node_id,omitempty"`
}

type UpdateJobBody struct {
	Image             *string `json:"image,omitempty"`
	Command           *string `json:"command,omitempty"`
	Schedule          *string `json:"schedule,omitempty"`
	ConcurrencyPolicy *string `json:"concurrency_policy,omitempty"`
	HistoryLimit      *int    `json:"history_limit,omitempty"`
}

func (c *Client) ListJobs(orgID, projectID string) ([]Job, error) {
	resp, err := c.do("GET", "/api/v1/orgs/"+orgID+"/projects/"+projectID+"/jobs", nil)
	if err != nil {
		return nil, err
	}
	return decode[[]Job](resp)
}

func (c *Client) CreateJob(orgID, projectID string, body CreateJobBody) (*Job, error) {
	resp, err := c.do("POST", "/api/v1/orgs/"+orgID+"/projects/"+projectID+"/jobs", body)
	if err != nil {
		return nil, err
	}
	return decodePtr[Job](resp)
}

func (c *Client) GetJob(orgID, projectID, jobID string) (*Job, error) {
	resp, err := c.do("GET", "/api/v1/orgs/"+orgID+"/projects/"+projectID+"/jobs/"+jobID, nil)
	if err != nil {
		return nil, err
	}
	return decodePtr[Job](resp)
}

func (c *Client) UpdateJob(orgID, projectID, jobID string, body UpdateJobBody) (*Job, error) {
	resp, err := c.do("PATCH", "/api/v1/orgs/"+orgID+"/projects/"+projectID+"/jobs/"+jobID, body)
	if err != nil {
		return nil, err
	}
	return decodePtr[Job](resp)
}

func (c *Client) TriggerJob(orgID, projectID, jobID string) (*JobRun, error) {
	resp, err := c.do("POST", "/api/v1/orgs/"+orgID+"/projects/"+projectID+"/jobs/"+jobID+"/trigger", nil)
	if err != nil {
		return nil, err
	}
	return decodePtr[JobRun](resp)
}

func (c *Client) DeleteJob(orgID, projectID, jobID string) error {
	return c.doNoContent("DELETE", "/api/v1/orgs/"+orgID+"/projects/"+projectID+"/jobs/"+jobID)
}

func (c *Client) ListJobRuns(orgID, projectID, jobID string) ([]JobRun, error) {
	resp, err := c.do("GET", "/api/v1/orgs/"+orgID+"/projects/"+projectID+"/jobs/"+jobID+"/runs", nil)
	if err != nil {
		return nil, err
	}
	return decode[[]JobRun](resp)
}

func (c *Client) DeleteJobRun(orgID, projectID, jobID, runID string) error {
	return c.doNoContent("DELETE", "/api/v1/orgs/"+orgID+"/projects/"+projectID+"/jobs/"+jobID+"/runs/"+runID)
}

func (c *Client) GetJobByName(orgID, projectID, ref string) (*Job, error) {
	jobs, err := c.ListJobs(orgID, projectID)
	if err != nil {
		return nil, err
	}
	for i, j := range jobs {
		if j.ID == ref || j.Name == ref {
			return &jobs[i], nil
		}
	}
	return nil, ErrNotFound("job", ref)
}

File: apps/web/src/lib/api/jobs.ts
import { apiFetch } from "./core"

export interface ApiJob {
  id: string
  project_id: string
  node_id: string | null
  name: string
  is_cron: boolean
  image: string
  command: string
  schedule: string
  concurrency_policy: string
  history_limit: number
  cpu_request: string
  cpu_limit: string
  memory_request: string
  memory_limit: string
  env_vars: string
  status: string
  last_run_at: string | null
  k8s_name: string
  created_at: string
  updated_at: string
}

export interface ApiJobRun {
  id: string
  job_id: string
  status: string
  started_at: string | null
  finished_at: string | null
  log: string
  k8s_job_name: string
  created_at: string
}

export interface CreateJobBody {
  name: string
  is_cron: boolean
  image: string
  command?: string
  schedule?: string
  concurrency_policy?: string
  history_limit?: number
  cpu_request?: string
  cpu_limit?: string
  memory_request?: string
  memory_limit?: string
  env_vars?: string
  node_id?: string
}

export const jobs = {
  list: (orgId: string, projectId: string, token: string) =>
    apiFetch<ApiJob[]>(`/api/v1/orgs/${orgId}/projects/${projectId}/jobs`, {}, token),

  get: (orgId: string, projectId: string, jobId: string, token: string) =>
    apiFetch<ApiJob>(`/api/v1/orgs/${orgId}/projects/${projectId}/jobs/${jobId}`, {}, token),

  create: (orgId: string, projectId: string, body: CreateJobBody, token: string) =>
    apiFetch<ApiJob>(
      `/api/v1/orgs/${orgId}/projects/${projectId}/jobs`,
      { method: "POST", body: JSON.stringify(body) },
      token
    ),

  update: (orgId: string, projectId: string, jobId: string, body: Partial<CreateJobBody>, token: string) =>
    apiFetch<ApiJob>(
      `/api/v1/orgs/${orgId}/projects/${projectId}/jobs/${jobId}`,
      { method: "PATCH", body: JSON.stringify(body) },
      token
    ),

  delete: (orgId: string, projectId: string, jobId: string, token: string) =>
    apiFetch<void>(
      `/api/v1/orgs/${orgId}/projects/${projectId}/jobs/${jobId}`,
      { method: "DELETE" },
      token
    ),

  listRuns: (orgId: string, projectId: string, jobId: string, token: string) =>
    apiFetch<ApiJobRun[]>(
      `/api/v1/orgs/${orgId}/projects/${projectId}/jobs/${jobId}/runs`,
      {},
      token
    ),

  trigger: (orgId: string, projectId: string, jobId: string, token: string) =>
    apiFetch<ApiJobRun>(
      `/api/v1/orgs/${orgId}/projects/${projectId}/jobs/${jobId}/trigger`,
      { method: "POST" },
      token
    ),

  deleteRun: (orgId: string, projectId: string, jobId: string, runId: string, token: string) =>
    apiFetch<void>(
      `/api/v1/orgs/${orgId}/projects/${projectId}/jobs/${jobId}/runs/${runId}`,
      { method: "DELETE" },
      token
    ),
}
